### PR TITLE
Rubric fixes

### PR DESCRIPTION
**Pull request by vargaat**
refs: MBL-15252
affects: Teacher
release note: Fixed grading issues when using rubrics in SpeedGrader.

test plan:
- See ticket.

In addition to that:
- Grade a submission with rubrics.
- Scroll two submissions away, then scroll back.
- Submission should still show your selected rubrics.
--
- Select a rubric on a submission, then de-select it.
- Close SpeedGrader, then re-open it.
- Rubric selection should be empty.
--
- Progress indicator circle should animate even if it's shown for less than 1 second.
--
- Rubric selection and comment should save upon adding them and not when the view is closed or the user scrolls to another submission.

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/69097683/p/420f5ed453d7b0608bf23b76c5cc9204">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/69097736/p/ae06fd4a348780ea9d5c5777c6c51d8d">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/69097627/p/878a5a97b7b91bccf505183d9fdf4d06">
    
  </a>
</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: XCTest failed: CoreTests/GradeSubmissionTests/testGrading</summary>


    Assertion Failure at GradeSubmissionTests.swift:50: XCTAssertEqual failed: ("nil") is not equal to ("Optional(false)")
</details>


[MBL-15252](https://instructure.atlassian.net/browse/MBL-15252)

#### Affected Apps: Teacher

#### Release Note: 
Fixed grading issues when using rubrics in SpeedGrader.
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/SwiftUIViews/ChangeObserver.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                self.onChange(of: value, perform: action)
+            } else {
+                ChangeObserver(value: value, action: action) {
+                    self
+                }
+            }
+        }
+    }
+}
+
```
Comment (balintbartok) [inline Core/Core/SwiftUIViews/ChangeObserver.swift:38]: since this is only for ios 13, we should add an @available(iOS, obsoleted: 14) as a reminder

**Review by balintbartok — APPROVED** [review]
QA + 1

File: Core/Core/SwiftUIViews/ChangeObserver.swift (hunk 1 above)
Comment (vargaat) [reply Core/Core/SwiftUIViews/ChangeObserver.swift:39]: Good idea, added.

**Review by balintbartok — APPROVED** [review]

**Review by idevainstr — APPROVED** [review]
reviewed

**Pull request merged**